### PR TITLE
Add identity user_account creation documentation

### PR DESCRIPTION
**Pull request by rileyanderson**
https://github.com/department-of-veterans-affairs/va.gov-team/issues/50827

[UserAccount_Creation_Flow.md](https://github.com/department-of-veterans-affairs/va.gov-team/blob/identity/user-accounts-documentation/products/identity/Design%20Documents/UserAccount_Creation_Flow.md)

[UserAccount_Attributes.md](https://github.com/department-of-veterans-affairs/va.gov-team/blob/identity/user-accounts-documentation/products/identity/Design%20Documents/UserAccount_Attributes.md)

@joeniquette is this what you were looking for? I wasn't sure how deep into the weeds I should go.

### CODE REVIEW

File: products/identity/Design Documents/UserAccount_Creation.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+## Summary
+`UserAccount` and `UserVerification` objects are created from two different methods, SSOe and SiS workflows.
+
+## UserAccount
+
+| Attribute | Description |
+| --- | --- |
+| `icn` | MPI Integration Control Number |
+
+### SSOe
```
Comment (asg5704) [inline products/identity/Design Documents/UserAccount_Creation.md:12]: Single Sign-On (SSOe)

File: products/identity/Design Documents/UserAccount_Creation.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+The `icn` is retreived from the `SAML` that is returned from IAM during the authentication flow. `UserAccount` objects are then found or created in after login actions.
+
+1. `V1::SessionsController#user_login` sets the `current_user` attributes (including `icn`) from the SAML and calls `V1::SessionsController#after_login_actions`.
+2. `V1::SessionsController#after_login_actions` creates and runs a new `Login::UserVerifier` service object with the current_user's `icn`.
+3. `Login::UserVerifier` service finds or creates a `UserAccount` with the current_user's `icn`.
+
+### SiS
```
Comment (asg5704) [inline products/identity/Design Documents/UserAccount_Creation.md:19]: Sign in Service (SiS)

**Review by asg5704 — CHANGES_REQUESTED** [review]
Looks good mostly, just a few tweaks 🚀

File: products/identity/Design Documents/UserAccount_Creation.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# UserAccount & UserVerification Creation
+
+## Summary
+`UserAccount` and `UserVerification` objects are created from two different methods, SSOe and SiS workflows.
```
Comment (bosawt) [inline products/identity/Design Documents/UserAccount_Creation.md:4]: 'created and maintained'

Comment (bosawt) [reply products/identity/Design Documents/UserAccount_Creation.md:4]: 'SSOe and SiS authentication workflows'

File: products/identity/Design Documents/UserAccount_Creation.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+1. `V1::SessionsController#user_login` sets the `current_user` attributes (including `icn`) from the SAML and calls `V1::SessionsController#after_login_actions`.
+2. `V1::SessionsController#after_login_actions` creates and runs a new `Login::UserVerifier` service object with the current_user's `icn`.
+3. `Login::UserVerifier` service finds or creates a `UserAccount` with the current_user's `icn`.
+
+### Sign in Service (SiS)
+The `icn` is retreived from querying MPI. If an MPI record does not exist, one is created. The `icn` is then used to find or create a `UserAccount`.
+
+1. `V0::SignInController#callback` calls `V0::SignInController#create_login_code`.
+2. `V0::SignInController#create_login_code` gets `verified_icn` from `SignIn::AttributeValidator` service object.
```
Comment (bosawt) [inline products/identity/Design Documents/UserAccount_Creation.md:23]: Here's the flow that happens during the sign in service flow:
- `SignInController#callback` -> `AttributeValidator`
- `AttributeValidator` attempts to query MPI based off the `idme_uuid`, `logingov_uuid`, or `mhv_icn` (depending on what the user has authenticated with)
  - If MPI record does not exist, `AttributeValidator` creates a new MPI record with the user's credential information
- ICN is returned to `SignInController`, etc, etc

File: products/identity/Design Documents/UserAccount_Creation.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## UserVerification
+A `UserAccount` can have many `UserVerifications`. There will be one `UserVerification` per IDP UUID.
+| Attribute | Description |
+| --- | --- |
+| `user_account_id` | Associated UserAccount |
+| `idme_uuid` | A users UUID from ID.me |
+| `logingov_uuid` | A users UUID from Login.gov |
+| `mhv_uuid` | A users UUID from My HealtheVet |
+| `dslogon_uuid` | A users UUID from DS Logon |
```
Comment (bosawt) [inline products/identity/Design Documents/UserAccount_Creation.md:36]: can you add `backing_idme_uuid`, which is relevant when a user authenticates with MHV or DS Logon, and it is the `idme_uuid` associated with either of these credentials, authenticated through ID.me

**Review by bosawt — COMMENTED** [review]
Most of the comments were just simple comments for clarify.

One additional piece of feedback would be that we may want a version of this document with an external audience in mind (basically, no code description, just descriptions of the `UserAccount`/`UserVerification` tables, with a description of each field and notes on how we get these values if necessary

**Comment by rileyanderson** [comment]
> Most of the comments were just simple comments for clarify.
> 
> One additional piece of feedback would be that we may want a version of this document with an external audience in mind (basically, no code description, just descriptions of the `UserAccount`/`UserVerification` tables, with a description of each field and notes on how we get these values if necessary

@bosawt Sounds good. I wasn't sure how deep into it or not we wanted to get. I can whip up another 'external' one.

**Pull request merged**